server/src/main/proto/spine/system/server/entity_history.proto

```diff
@@ -0,0 +1,172 @@
+import "google/protobuf/timestamp.proto";
+
+import "spine/core/command.proto";
+import "spine/core/event.proto";
+import "spine/client/entities.proto";
+import "spine/server/entity/entity.proto";
+
+// An identifier of an entity history.
+//
+message EntityHistoryId {
+
+    // The ID of the entity.
+    spine.client.EntityId entity_id = 1;
+
+    // The type URL of the entity state.
+    string type_url = 2;
+}
+
+// A history of interactions with an entity.
+//
+// Each Aggregate, Projection, and ProcessManager instance has a corresponding entity history.
+//
+// Stores the lists of all commands and events ever dispatched to the entity, represented by their
+// IDs.
+//
+message EntityHistory {
+    option (entity).kind = AGGREGATE;
+
+    EntityHistoryId id = 1;
+
+    // The kind of the entity.
+    EntityOption.Kind kind = 2;
+
+    // The history of lifecycle flags changes.
+    LifecycleHistory lifecycle = 3;
+
+    // The most recent time when the entity state has been changed.
+    google.protobuf.Timestamp last_state_change = 4;
+
+    // The history of messages dispatched to the entity.
+    DispatchingHistory dispatching = 5;
+}
+
+// The history of an entity lifecycle.
+//
+// Contains the current `LifecycleFlags` and the timestamps of the instants when
+// the lifecycle-changing mutations have occured to the entity for the most recent time.
+//
+// Since none of the lifecycle states is terminal, none of the fields are mutually exclusive.
+//
+message LifecycleHistory {
+
+    // Current `LifecycleFlags` of the entity.
+    spine.server.entity.LifecycleFlags lifecycle_flags = 1;
+
+    // Instant when the entity was archived.
+    //
+    // If the entity has been archived for multiple times in its history, this timestamp represents
+    // the most recent occurance.
+    //
+    google.protobuf.Timestamp when_archived = 2;
+
+    // Instant when the entity was deleted.
+    //
+    // If the entity has been deleted for multiple times in its history, this timestamp represents
+    // the most recent occurance.
+    //
+    google.protobuf.Timestamp when_deleted = 3;
+
+    // Instant when the entity was extracted from archive.
+    //
+    // If the entity has been extracted for multiple times in its history, this timestamp represents
+    // the most recent occurance.
+    //
+    google.protobuf.Timestamp when_extracted_from_archive = 4;
+
+    // Instant when the entity was restored after deletion.
+    //
+    // If the entity has been restored for multiple times in its history, this timestamp represents
+    // the most recent occurance.
+    //
+    google.protobuf.Timestamp when_restored = 5;
+}
+
+// The history of dispatching messages to the entity.
+//
+// Contains timespamps of most recent command and event dispatched to the entity.
+//
+message DispatchingHistory {
+
+    // The most recent time when a command was dispatched to the entity.
+    google.protobuf.Timestamp last_command = 1;
```

Comment:
`when_command` and `when_event`.

`getLast_*` is easily confused with the methods that actually return the last command and the last event.